src/lmdb.ml

```diff
@@ -58,7 +58,13 @@ module Env = struct
   end
 
   let create ?max_readers ?map_size ?max_dbs ?(flags=Flags.none) ?(mode=0o755) path =
-    let mode = coerce uint32_t mode_t UInt32.(of_int mode) in
+    let mode =
+      (* Happens on OSX.. *)
+      if (sizeof mode_t) = 2 then
+        coerce uint16_t mode_t UInt16.(of_int mode)
+      else
+        coerce uint32_t mode_t UInt32.(of_int mode)
+    in
```

Comment:
Yes, that's what I think. When I first wrote this, your library wasn't around, but I have no excuses anymore. :)